### PR TITLE
Update the browser module

### PR DESCRIPTION
**Pull request by coolreader18**
cc @phorward

Main relevant changes:

- Move `Promise` to the `_js` module
- Rename `JsValue`/`JsError` to `JSValue`/`JSError` to fit with Python naming conventions
- If an exception bubbles up to js and it's a `JSError`, throw the jsvalue inside it
- `browser.jsclosure -> JSClosure`
```py
class JSClosure:
    value: Optional[JSValue]
    destroyed: bool
    detached: bool
    destroy(self)
    detach(self) -> JSValue
```
`JSClosure.destroy()` decreases the refcount of the inner python function that gets called, and makes the js function uncallable. `JSValue.detach()` makes it so the inner function never gets dropped. If anyone has any API critiques, I'd love to hear them; I'm not really sure if having `destroyed` and `detached` completely orthogonal to each other is the best way to do things.

### CODE REVIEW

File: vm/src/builtins/mod.rs [hunk 1]
```diff
@@ -29,6 +29,7 @@ pub use function::PyFunction;
 pub(crate) mod generator;
 pub use generator::PyGenerator;
 pub(crate) mod getset;
+pub use getset::PyGetSet;
```
Comment (youknowone) [inline vm/src/builtins/mod.rs:32]: oh, i thought this is internally used only but not in this case. maybe we'd better to expose builtinfunc also.

**Review by youknowone — APPROVED** [review]

File: wasm/lib/src/browser_module.rs [hunk 2]
```diff
@@ -361,11 +263,10 @@ pub fn make_module(vm: &VirtualMachine) -> PyObjectRef {
 
     let element = Element::make_class(ctx);
 
-    py_module!(vm, "browser", {
+    py_module!(vm, "_browser", {
```
Comment (phorward) [inline wasm/lib/src/browser_module.rs:266]: Ok this allows now for an `import _browser` in WASM.

But what are the differences between the browser (frozen) and the browser (built-in) module, they don't seem to be the same:
```
>>>> dir(browser)
['Document', 'Element', 'JSValue', 'Promise', '__annotations__', '__builtins__', '__doc__', '__loader__', '__name__', '__package__', '__spec__', '_alert', '_confirm', '_prompt', 'alert', 'cancel_animation_frame', 'confirm', 'document', 'fetch', 'jsclosure', 'jsstr', 'load_module', 'prompt', 'request_animation_frame', 'window']
>>>>> dir(_browser)
['Document', 'Element', '__doc__', '__loader__', '__name__', '__package__', '__spec__', 'cancel_animation_frame', 'document', 'fetch', 'load_module', 'request_animation_frame']
>>>>> browser
<module 'browser' (frozen)>
>>>>> _browser
<module '_browser' (built-in)>
```

Comment (coolreader18) [reply wasm/lib/src/browser_module.rs:266]: It doesn't actually change anything but `import _browser; _browser.__name__`, if you look at the source of `browser.py` you can see that it imports `*` from `_browser`. `_browser` has stuff like fetch and document, `browser` wraps it and also provides some other functions that are easy to implement in Python. It's like the built-in `posix`/`nt` vs the Python `os`.

Comment (coolreader18) [reply wasm/lib/src/browser_module.rs:266]: [For reference](https://github.com/RustPython/RustPython/blob/master/wasm/lib/src/browser.py)
